src/lib.rs

```diff
@@ -33,6 +33,33 @@ use std::fmt;
 use std::io;
 use std::ops;
 
+macro_rules! convert_u8_to_u32 {
+    (ref mut $count:expr, $data:expr) => {
+        unsafe { std::mem::transmute::<&mut [u8; $count], &mut [u32; $count / 4]>($data) }
+    };
+}
+
+macro_rules! convert_u32_to_u8 {
+    (ref $count:expr, $data:expr) => {
+        unsafe { std::mem::transmute::<&[u32; $count], &[u8; 4 * $count]>($data) }
+    };
+    ($count:expr, $data:expr) => {
+        unsafe { std::mem::transmute::<[u32; $count], [u8; 4 * $count]>($data) }
+    };
+}
```

Comment:
I'm not really happy with the transmutes at all. There must be a better thing. Maybe https://stackoverflow.com/a/29065350 ?

Reply:
Same goes for here with the endianess